sympy/sets/conditionset.py

```diff
@@ -246,6 +246,37 @@ def _eval_subs(self, old, new):
             return ConditionSet(new, Contains(new, base), base)
         return self.func(self.sym, cond, base)
 
+    def _as_interval(self):
+        """`as_interval` is only implemented for univariate conditions and real base sets"""
+        if not isinstance(self.sym, Tuple) and  self.base_set.is_subset(S.Reals):
+            try:
+                return self.condition.as_set().intersect(self.base_set)
+            except NotImplementedError:
+                return self
```

Comment:
Can you give an example of when NotImplementedError would be raised here? Which part of the expression raises? Is it possible to only catch the exception from the relevant part of the expression?

Reply:
`as_set` is an expensive routine calling the solver `solve_univariate_inequality`, and I guess `NotImplementedError` can potentially be raised from there.